pkg/api/types.go

```diff
@@ -1776,6 +1774,110 @@ type SecretList struct {
+	// and volumes
+	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+
+	// RunAsUser is the UID to run the entrypoint of the container process.  Corresponding option is --user or -u
+	RunAsUser int64 `json:"runAsUser,omitempty"`
+}
+
+// SELinuxOptions are the labels to be applied to the container
+type SELinuxOptions struct {
+	// User --security-opt="label:user:USER"
+	User string `json:"user,omitempty"`
+
+	// Role --security-opt="label:role:ROLE"
+	Role string `json:"role,omitempty"`
+
+	// Type --security-opt="label:type:TYPE"
+	Type string `json:"type,omitempty"`
+
+	// Level --security-opt="label:level:LEVEL"
+	Level string `json:"level,omitempty"`
+
+	// Disabled --security-opt="label:disable"
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+// SecurityConstraints provides the constraints that at security context provider will
+// ensure that applied SecurityContext requests follow.  When a setting is provided in the SecurityConstraints
+// that conflicts with an actual request it will be implementation specific whether that request is
+// ignored and the container is still run (without the requested constraint) or if the container will be failed
+type SecurityConstraints struct {
+	// EnforcementPolicy will drive behavior for how the constraints are enforce
+	EnforcementPolicy SecurityConstraintPolicy `json:"enforcementPolicy,omitempty"`
+
+	// AllowPrivileged indicates whether this context allows privileged mode containers
+	AllowPrivileged bool `json:"allowPrivileged,omitempty"`
+
+	// SELinux provides the security constraint options for selinux
+	SELinux *SELinuxSecurityConstraints `json:"seLinux,omitempty"`
+
+	// AllowCapabilities dictates if a container can request to add or drop capabilites
+	AllowCapabilities bool `json:"allowCapabilities,omitempty"`
+
+	// AllowCapabilities dictates if a container can request to run the entry point process as a specific user
+	AllowRunAsUser bool `json:"allowRunAsUser,omitempty"`
+
+	// Capabilities represents, if AllowCapabilities is true, the caps that requests
+	// are allowed to add or drop.
+	Capabilities *Capabilities `json:"capabilities,omitempty"`
+
+	// DefaultSecurityContext is applied to any container that does not have a security context set.  It must
+	// also conform to the constraints defined in SecurityConstraints object
+	DefaultSecurityContext *SecurityContext `json:"defaultSecurityContext,omitempty"`
+
+	// List of pod sources for which using host network is allowed.
+	HostNetworkSources []string
+}
+
+// SELinuxSecurityConstraints defines what is allowed in SecurityContext requests with regards to SELinux label options
+// that are currently supported by docker
+type SELinuxSecurityConstraints struct {
+	// AllowUserLabel --security-opt="label:user:USER"
+	AllowUserLabel bool `json:"allowUserLabel,omitempty"`
+
+	// AllowRoleLabel --security-opt="label:role:ROLE"
+	AllowRoleLabel bool `json:"allowRoleLabel,omitempty"`
```

Comment:
All of these have complex constraints that can't be captured with booleans (i.e. this doesn't look like it's being driven by the concrete requirements in the security doc).  Can you include the options necessary to satisfy the requirements?